ethcore/private-tx/src/lib.rs

```diff
@@ -683,7 +686,7 @@ impl ChainNotify for Provider {
 	fn new_blocks(&self, imported: Vec<H256>, _invalid: Vec<H256>, _route: ChainRoute, _sealed: Vec<H256>, _proposed: Vec<Bytes>, _duration: Duration) {
 		if !imported.is_empty() {
 			trace!("New blocks imported, try to prune the queue");
-			if let Err(err) = self.process_queue() {
+			if let Err(err) = self.process_verification_queue() {
```

Comment:
Ok, so it seems some transactions will become ready after a block, right? Makes sense to keep the `process_verification_queue` as a separate function then, but we need to fix the error handling there.